monitor/src/external_api.rs

```diff
@@ -0,0 +1,154 @@
+use std::path::Path;
+use crate::config::{Config, SingleConfig};
+use actix_web::{web, HttpResponse, Responder, Result};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+pub struct AppState {
+    pub(crate) configs: Arc<RwLock<Config>>,
+}
+
+#[derive(Serialize)]
+struct ConfigWithId {
+    id: usize,
+    event: String,
+    repo_owner: String,
+    repo_name: String,
+    github_token: String,
+    actions_path: String,
+}
+
+pub async fn get_configurations(data: web::Data<AppState>) -> impl Responder {
+    let configs = data.configs.read().await;
+
+    let configs_with_id: Vec<ConfigWithId> = configs
+        .configurations
+        .iter()
+        .enumerate()
+        .map(|(index, config)| ConfigWithId {
+            id: (index + 1),
+            event: config.event.clone(),
+            repo_owner: config.repo_owner.clone(),
+            repo_name: config.repo_name.clone(),
+            github_token: config.github_token.clone(),
+            actions_path: config.actions_path.clone(),
+        })
+        .collect();
+
+    HttpResponse::Ok().json(configs_with_id)
+}
+
+pub async fn get_configuration_by_id(
+    data: web::Data<AppState>,
+    path: web::Path<usize>,
+) -> Result<impl Responder> {
+    let id = path.into_inner();
+    let configs = data.configs.read().await;
+
+    if id >= 1 && id <= configs.configurations.len() {
+        let config = &configs.configurations[id - 1];
+        Ok(HttpResponse::Ok().json(config))
+    } else {
+        Ok(HttpResponse::NotFound().body("Configuration not found"))
+    }
+}
+
+pub async fn get_actions_file(
+    data: web::Data<AppState>,
+    path: web::Path<usize>,
+) -> Result<impl Responder> {
+    let id = path.into_inner();
+    let configs = data.configs.read().await;
+
+    if id >= 1 && id <= configs.configurations.len() {
+        let config = &configs.configurations[id - 1];
+        let actions_file_content = std::fs::read_to_string(&config.actions_path)
+            .unwrap_or_else(|_| "Actions file not found".to_string());
+        Ok(HttpResponse::Ok().body(actions_file_content))
+    } else {
+        Ok(HttpResponse::NotFound().body("Configuration not found"))
+    }
+}
+
+#[derive(Deserialize)]
+pub struct NewConfig {
+    event: String,
+    repo_owner: String,
```

Comment:
Is it more interesting to do that or use directly a repo url ? 
I say that because the agent will need it later and it would be easier to handler later. But if it is well handled later it is ok for me.

Reply:
To me, as the repo_url is always computable using the repo_name and the repo_user, I would rather keep this as it is already.